contrib/generate-emulation.py

```diff
@@ -121,16 +163,35 @@ def _get_host_devices_and_attrs() -> str:
     return generator.to_data()[0]
 
 
-if len(sys.argv) < 2:
-    sys.stdout.write(_minimize_json(sys.stdin.read()))
-else:
-    for fn in sys.argv[1:]:
-
-        try:
-            with open(fn, "rb") as f_in:
-                json_in = f_in.read().decode()
-        except FileNotFoundError:
-            json_in = _get_host_devices_and_attrs()
-        json_out = _minimize_json(json_in).encode()
-        with open(fn, "wb") as f_out:
-            f_out.write(json_out)
+def main():
+    if len(sys.argv) < 2:
+        sys.stdout.write(_minimize_json(sys.stdin.read()))
+    else:
+        parser = argparse.ArgumentParser(description='Generate HSI emulation json.')
+        parser.add_argument('json_file_list', metavar='json_file_list', type=str, nargs='+',
+                    help='An list of HSI emulation json file.')
+        parser.add_argument('--machine', default="physical",
+                    help='Machines kind (default: physical)')
+        parser.add_argument('--chassis', default="laptop",
+                    help='Chassis type (laptop and desktop. default: physical)')
```

Comment:
Can't we get this information out of fwupd so you don't need to override it on the command line for generation of emulation data?  I would think it makes more sense to populate it through the daemon daemon so it can export it.

Reply:
I think most developers will run fwupd in a VM to develop and test their HSI-related software. They need to set both to emulate the environment, for example, HSI hypervisor exception, HSI level 0-4...etc. These arguments can actually simplify the process to generate emulation data without editing the JSON again.